packages/expo-background-task/android/src/main/java/expo/modules/backgroundtask/BackgroundTaskConsumer.kt

```diff
@@ -0,0 +1,116 @@
+package expo.modules.backgroundtask
+
+import android.content.Context
+import android.util.Log
+import com.facebook.react.common.build.ReactBuildConfig
+import expo.modules.core.errors.ModuleDestroyedException
+import expo.modules.core.interfaces.LifecycleEventListener
+import expo.modules.interfaces.taskManager.TaskConsumer
+import expo.modules.interfaces.taskManager.TaskConsumerInterface
+import expo.modules.interfaces.taskManager.TaskExecutionCallback
+import expo.modules.interfaces.taskManager.TaskInterface
+import expo.modules.interfaces.taskManager.TaskManagerUtilsInterface
+import expo.modules.interfaces.taskManager.TaskServiceProviderHelper
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.launch
+
+const val BACKGROUND_TASK_TYPE = "expo-background-task"
+const val DEFAULT_INTERVAL_MINUTES = 60L * 24L // Once every day
+private val TAG = BackgroundTaskConsumer::class.java.simpleName
+
+class BackgroundTaskConsumer(context: Context?, taskManagerUtils: TaskManagerUtilsInterface?) :
+  TaskConsumer(context, taskManagerUtils), TaskConsumerInterface, LifecycleEventListener {
+
+  private var task: TaskInterface? = null
+  private var isBackgrounded: Boolean = true
+  private val taskCoroutineScope = CoroutineScope(Dispatchers.Default)
+
+  override fun taskType(): String {
+    return BACKGROUND_TASK_TYPE
+  }
+
+  /**
+   * Exposing the execute task function so that the BackgroundTaskWork (WorkManager Work unit)
+   * can execute the inner task. A task is only executed if the app is in the background.
+   * This only applies when the app is in release mode.
+   */
+  fun executeTask(callback: TaskExecutionCallback) {
+    if (isBackgrounded || ReactBuildConfig.DEBUG) {
```

Comment:
i'm not quite clear for this. the comment said "This only applies when the app is in release mode." but the code is checking for debug build. could you help to explain more?

Reply:
I will refactor this a bit - I agree with you.